lxc/storage_volume.go

```diff
@@ -963,7 +966,10 @@ func (c *cmdStorageVolumeDetachProfile) run(cmd *cobra.Command, args []string) e
 	// Find the device
 	if devName == "" {
 		for n, d := range profile.Devices {
-			if d["type"] == "disk" && d["pool"] == resource.name && d["source"] == args[1] {
+			volType, volName := parseVolume("custom", args[1])
+			sourceType, sourceName := parseVolume("custom", d["source"])
```

Comment:
While I see the purpose of parsing the source to be safe, I don't think the source name is ever supposed to be the full volume name.
And if it somehow show as type/name it would be nice for it to be flagged right away so we can fix it, so in my view we can drop this second `parseVolume`.

Reply:
Oh I see, sow the type will be included because of `virtual-machine` volumes. You can disregard this.